Integrate docs checkers into the noxfile (#668)
ansible/ansible-documentation: README.md

### Running automated tests

This project includes a `nox` configuration to automate tests, checks, and other functions.
You can use these automated tests to help you verify changes before you submit a PR.

1. Install `nox` using `python3 -m pip install nox` or your distribution's package manager.
2. Run `nox --list` from the repository root to view available sessions.

Each `nox` session creates a temporary environment that installs all requirements and runs the test or check.
This means you only need to run one command to perform the test accurately and consistently.
Here are some `nox` sessions you can run:

* Run `nox` with no arguments to run all available sessions.
* Run `nox -s clone-core` to clone required parts of the `ansible/ansible` repository as mentioned in the [local build documentation](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally).

> * Run `nox -s "checkers(rstcheck)"` to ensure there are no syntax errors in the reStructuredText source files.

Author: The `rstcheck` docs checker also supports `--relaxed`, by the way, so we should mention it in both places if you insist on keeping it in the next bullet point.

Reviewer: I don't insist on keeping it there but figured it was mostly relevant for the docs build so didn't see the need to mention it with rstcheck.

> * Run `nox -s "checkers(rstcheck)"` to ensure there are no syntax errors in the reStructuredText source files.

Author: I would also link to the rstcheck section of the documentation_contributors page.

> * Run `nox -s "checkers(docs-build)"` or `nox -s "checkers(docs-build)" -- --relaxed` to verify the docs build.

Author: I would not mention the `--relaxed` thing for now. It's moreso there for testing newer antsibull and sphinx there and most contributors shouldn't need to concern themselves with it.

> * Run `nox -s "checkers(docs-build)"` or `nox -s "checkers(docs-build)" -- --relaxed` to verify the docs build.

Author: ```suggestion
* Run `nox -s "checkers(docs-build)"` or `nox -s "checkers(docs-build)" -- --relaxed` to verify the docs build.
* Run `nox -s lint` to lint, type check, and format the python scripts in this repository.
```

Reviewer: I've included this suggestion but won't commit directly.

> The `nox -s "checkers(docs-build)"` session cleans the generated docs after it runs.

> > If you want to view the generated HTML in your browser, you should build the documentation locally.

Author: I'd also link to the appropriate section of the documentation_contributors page here.

## Checking spelling

Use [`codespell`](https://github.com/codespell-project/codespell) to check for common spelling mistakes in the documentation source.

* Run `nox -s spelling` to check spelling.
* Run `nox -s spelling -- -w` to correct spelling errors.

> > When `codespell` suggests more than one word as a correction, run `nox -s spelling -- -w -i 3` to select an option.

Author: I think it would make sense to include this in the bulleted list instead of as a separate blockquote, but that's just me.

Reviewer: Yeah that does make sense. Thanks.

## Updating dependencies

Run `nox -e pip-compile` to update project dependency versions.

> This session requires Python 3.10.

If you do not have Python 3.10 installed, you can use root-less podman with a Python 3.10 image as follows:

```bash

> podman run --rm --tty --volume "$(pwd):/mnt:z" --workdir /mnt python:3.10 bash -c 'pip install nox ; nox -e pip-compile'

Author: While we're at it, I'd like to change this to use the fully qualified container name. If you prefer to handle this in a separate PR, that's fine with me as well.

```suggestion
podman run --rm --tty --volume "$(pwd):/mnt:z" --workdir /mnt docker.io/library/python:3.10 bash -c 'pip install nox ; nox -e pip-compile'
```

Reviewer: I think it's fine to do that here.

Follow the documentation to [set up your environment](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally) and then [build Ansible community documentation locally](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#building-the-documentation-locally)

## Testing your changes

We welcome all contributions to Ansible community documentation.
If you plan to submit a pull request with changes, you should first run tests from this repository.

> Doing so ensures that your pull request does not trigger any ci failures.

Author: ```suggestion
Doing so ensures that your pull request does not trigger any CI failures.
```

## Preparing to run tests

This project includes a `nox` configuration to automate tests, checks, and other functions.

1. Install `nox` using `python3 -m pip install nox` or your distribution's package manager.
2. Run `nox --list` from the repository root to view available sessions.

Each `nox` session creates a temporary environment that installs all requirements and runs the test or check.
This means you only need to run one command to perform the test accurately and consistently.

## Running all tests

Run `nox` with no arguments to run all available sessions.

> ## Running the docs-build session

Author: ```suggestion
## Running the docs-build checker
```

At some point, we'll add session to do a regular docs build, so let's specify that this is for the **checker**.

Ensure your changes do not introduce errors and the docs build succeeds.

> 1. Clone parts of the `ansible/ansible` repository with `nox -s "checkers(docs-build)"`.

Author: ```suggestion
1. Clone parts of the `ansible/ansible` repository with `nox -s clone-core`.
```

## Testing your changes

We welcome all contributions to Ansible community documentation.
If you plan to submit a pull request with changes, you should first run tests from this repository.
Doing so ensures that your pull request does not trigger any ci failures.

## Preparing to run tests

This project includes a `nox` configuration to automate tests, checks, and other functions.

1. Install `nox` using `python3 -m pip install nox` or your distribution's package manager.
2. Run `nox --list` from the repository root to view available sessions.

Each `nox` session creates a temporary environment that installs all requirements and runs the test or check.
This means you only need to run one command to perform the test accurately and consistently.

## Running all tests

Run `nox` with no arguments to run all available sessions.

## Cloning required parts of the Ansible core repository

> As mentioned in the [local build documentation](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally) you should periodically clone required parts of the `ansible/ansible` repository.

Author: ```suggestion
As mentioned in the [local build documentation](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally), you should periodically clone required parts of the `ansible/ansible` repository.
```
